### DISCUSSION TITLE
Exception using word_timestamps = True

### DISCUSSION THREAD

**Discussion by chenkaigithub** [openai/whisper#2464] [Q&A]
When I use  word_timestamps = True in transcribe function. An exception
'<code object dtw at 0x0000020163D5C200, file "\\venv\\Lib\\site-packages\\whisper\\timing.py", line 141> != <code object dtw_cpu at 0x0000020161EC9C40, file "\\venv\\Lib\\site-packages\\whisper\\timing.py", line 82>' is shown, how can I sovle this? I only use whisper.load_model("base",device="cpu")  Thank you.

**Comment by DidouJazz** [comment]
This is the code I am using:
    # Set the environment variable to point to the new model directory
    os.environ["WHISPER_MODEL_DIR"] = "whisper"

    device = "cuda" if torch.cuda.is_available() else "cpu"
    model = whisper.load_model(model_name).to(device)
    language = "en" if model_name.endswith(".en") else None

    result = model.transcribe(video_file, language=language, fp16=False, word_timestamps=True)
I am running python 3.12.8 on WIndows and here is the excepition : 
Exception has occurred: AssertionError
<code object dtw at 0x000001A31B49EF50, file "XXX\.venv_win\Lib\site-packages\whisper\timing.py", line 141> != <code object dtw_cpu at 0x000001A31B0586D0, file "XXX\.venv_win\Lib\site-packages\whisper\timing.py", line 82>
KeyError: (82, 'XXX\\.venv_win\\Lib\\site-packages\\whisper\\timing.py', <code object dtw_cpu at 0x000001A31B0586D0, file "XXX\.venv_win\Lib\site-packages\whisper\timing.py", line 82>)

During handling of the above exception, another exception occurred:

  File "XXX\futur\src\YYY.py", line 652, in transcribe_video
    result = model.transcribe(video_file, language=language, fp16=False, word_timestamps=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "XXX\futur\src\import whisper.py", line 28, in <module>
    whisper_output = YYY.transcribe_video(f"{path}\\ddprod\\{name}.mp4")
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: <code object dtw at 0x000001A31B49EF50, file "XXX\.venv_win\Lib\site-packages\whisper\timing.py", l

**Reply by DidouJazz** [reply]
My use-case is to replicate this:
whisper "XXX\ddprod\YYY.mp4" `
    --output_dir "XXX\ddprod" `
    --model tiny `
    --model_dir whisper `
    --output_format vtt `
    --max_line_width 32 `
    --max_line_count 2 `
    --word_timestamps True
   This is working as expected

**Comment by DidouJazz** [comment]
Name: openai-whisper
Version: 20240930
Summary: Robust Speech Recognition via Large-Scale Weak Supervision
Home-page: https://github.com/openai/whisper
Author: OpenAI
Author-email:
License: MIT
Location: C:\Users\fradae\OneDrive - SAS\SAS\GEL\GitLab\gelpowervid\.venv_win\Lib\site-packages
Requires: more-itertools, numba, numpy, tiktoken, torch, tqdm
Required-by:

**Comment by DidouJazz** [comment]
Here is a simple example in order to reproduce:
```python
import whisper
from whisper.utils import get_writer
import torch

def transcribe_audio(audio_path, max_line_width=None, max_line_count=None):
    # Load the Whisper model (using tiny.en on CPU)
    model = whisper.load_model("tiny.en", device="cpu")
    
    # Ensure we're using CPU for processing
    torch.set_grad_enabled(False)
    if torch.cuda.is_available():
        torch.cuda.empty_cache()
    
    # Transcribe the audio with specified parameters
    result = model.transcribe(
        audio_path,
        language="en",
        fp16=False,  # Ensure we're not using fp16 on CPU
        verbose=True,
        word_timestamps=True
    )
    
    # Set up writer options
    writer_options = {
        "max_line_count": max_line_count,
        "max_line_width": max_line_width
    }
    
    # Get SRT writer and write to current directory
    output_directory = "./"
    writer = get_writer("srt", output_directory)
    
    # Write the result with specified formatting
    writer(result, audio_path, writer_options)
    
    return result["text"]

if __name__ == "__main__":
    # Example usage
    audio_file = r"C:\Users\fradae\Downloads\01_010 Workshop Introduction.mp4"
    
    try:
        # Example with both parameters
        transcription = transcribe_audio(
            audio_file,
            max_line_width=32,    # One word per line
            max_line_count=3         # Maximum number of lines
        )
        
        print("\nTranscription has been saved as an SRT file in the current directory")
        print("\nFull transcription text:")
        print(transcription)
    except Exception as e:
        print(f"An error occurred during transcription: {str(e)}")
        print("Full error details:", e.__class__.__name__)
```
